### DISCUSSION TITLE
Resuming an interrupted run

### DISCUSSION THREAD

**Discussion by ah-dinh** [Parcels-code/Parcels#1357] [Q&A] [answered]
Hi, 

Is there a way to periodically save the progress of a run and resume it at a later time? I'm doing large runs over the span of many weeks that sometimes get interrupted due to system shutdown and other issues. The only solution I can think of is breaking the long run into shorter segments and re-initialize the particles from their previous location. Before I attempt to implement that, is there a built-in checkpoint/recovery method or an easier work around for me to resume an interrupted run?

**Comment by erikvansebille** [comment, marked as answer]
Yes, you can use the `particleset.from_particlefile()` method, see [here](https://parcels.readthedocs.io/en/latest/reference/particles.html?highlight=from_particlefile#parcels.particleset.baseparticleset.BaseParticleSet.from_particlefile) for the API reference, and the unit test below for an example implementation https://github.com/OceanParcels/parcels/blob/5578e4bd7863be620eac6fefbb12939755959a11/tests/test_particle_sets.py#L85-L115
